### PR TITLE
[JENKINS-65806] Correct handling of `orclIsEnabled` for Oracle Internet Directory

### PR DESCRIPTION
**Pull request by mattheimer**
This change corrects the handling of the orclIsEnabled attribute to detect disabled accounts for Oracle Internet Directory (OID). This issue was originally reported in 2021 under <https://issues.jenkins.io/browse/JENKINS-65806>

First the issue was confirmed to still occur using the most recent LDAP plugin. When attempting to configure Jenkins to use LDAP with Oracle Internet Directory the Test LDAP Settings results will  show Login success and be able to list some user attributes and groups but report failures for Lookup and Lockout. The output is:

    User lookup: failed for user "matt.heimer@oracle.com"
    The user "matt.heimer@oracle.com" is administratively disabled.

    The user "matt.heimer@oracle.com" may be unable to login with API tokens or SSH keys.
    The user will have inconsistent permissions if able to login using API tokens or SSH keys!
    If this is your own account this could mean you may be locked out!
    Are you sure you want to save this configuration?

The most recent OID documentation states what the treatment of orclisenabled should be https://docs.oracle.com/en/middleware/idm/internet-directory/12.2.1.4/administer/managing-accounts-and-passwords-oracle-internet-directory.html#GUID-396DFEB3-C09A-4435-B7B0-64F69384A0F9

> To permanently disable the account, set the orclisenabled attribute to DISABLED. Setting this attribute to any other value enables the account.

The plugin code was checking if that value was not equal to `enabled`, this change corrects the behavior to checking if it is equal to `disabled`. After the change LDAP integration with OID works correctly and testing reports `User lookup: successful`.

The README.md was also updated to mention support for the `orclIsEnabled` attribute in the Administratively Disabled Accounts section.

```[tasklist]
### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Review by jtnord — APPROVED** [review]
previous versions documentation at https://docs.oracle.com/middleware/11119/oid/administer/oid_susers.htm#OIDAG2238

> To permanently disable the account, set the orclisenabled attribute to DISABLED. Setting this attribute to any other value enables the account.

File: README.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ to indicate such. The following attributes are all supported:
 * `msDS-UserAccountDisabled` value of `TRUE`: modern [Active Directory attribute](https://docs.microsoft.com/en-us/windows/win32/adschema/a-msds-useraccountdisabled);
 * `userAccountControl` (UAC) or `msDS-User-Account-Control-Computed` with bit flag of `ADS_UF_ACCOUNTDISABLE` (0x2) present; and
 * `loginDisabled` value of `TRUE`: eDirectory attribute.
+* `orclIsEnabled` value of `DISABLED`: Oracle Internet Directory
```
Comment (daniel-beck) [inline README.md:40]:
```suggestion
* `loginDisabled` value of `TRUE`: eDirectory attribute;
* `orclIsEnabled` value of `DISABLED`: [Oracle Internet Directory attribute](https://docs.oracle.com/en/middleware/idm/internet-directory/12.2.1.4/administer/managing-accounts-and-passwords-oracle-internet-directory.html#GUID-2E7FAE4E-9762-4559-8204-776A1BF5E66C).
```

Weird list format, and adding a reference to docs.

File: README.md [hunk 2]
```diff
@@ -36,7 +36,8 @@ to indicate such. The following attributes are all supported:
 * `pwdAccountLockedTime` value of `000001010000Z`: common LDAP attribute using password policy overlay;
 * `msDS-UserAccountDisabled` value of `TRUE`: modern [Active Directory attribute](https://docs.microsoft.com/en-us/windows/win32/adschema/a-msds-useraccountdisabled);
 * `userAccountControl` (UAC) or `msDS-User-Account-Control-Computed` with bit flag of `ADS_UF_ACCOUNTDISABLE` (0x2) present; and
```
Comment (daniel-beck) [inline README.md:38]:
```suggestion
* `userAccountControl` (UAC) or `msDS-User-Account-Control-Computed` with bit flag of `ADS_UF_ACCOUNTDISABLE` (0x2) present;
```

File: README.md [hunk 3]
```diff
@@ -36,7 +36,8 @@ to indicate such. The following attributes are all supported:
...
-* `loginDisabled` value of `TRUE`: eDirectory attribute.
+* `loginDisabled` value of `TRUE`: eDirectory attribute;
```
Comment (daniel-beck) [inline README.md:39]:
```suggestion
* `loginDisabled` value of `TRUE`: eDirectory attribute; and
```

**Review by daniel-beck — COMMENTED** [review]
My previous suggestion was incomplete, list format is weirder than expected.

(Unfortunately, GH blocks creation a single suggestion over two lines.)

**Comment by mattheimer** [comment]
Hi @jtnord We can go all the way back to 2009, https://docs.oracle.com/cd/E12839_01/admin.1111/e10029/oid_susers.htm#BABBGBAB and the documented treatment of orclisenabled has remained unchanged. 

> To permanently disable the account, set the orclisenabled attribute to DISABLED. Setting this attribute to any other value enables the account.

**Pull request merged**